frontend/packages/dev-console/console-extensions.json

```
@@ -996,6 +997,74 @@
       "insertBefore": "devconsole.routingOptions"
     },
     "flags": {
+      "required": ["KNATIVE_SERVING_SERVICE", "OPENSHIFT_DEPLOYMENTCONFIG"]
+    }
+  },
+  {
+    "type": "console.user-preference/item",
+    "properties": {
+      "id": "devconsole.preferredResource",
+      "label": "%devconsole~Resource type%",
+      "groupId": "applications",
+      "description": "%devconsole~If resource type is not selected, the console default to the latest.%",
+      "field": {
+        "type": "dropdown",
+        "userSettingsKey": "devconsole.preferredResourceType",
+        "defaultValue": "kubernetes",
+        "description": "%devconsole~The defaults below will only apply to the Import from Git and Deploy Image forms when creating Deployments or Deployment Configs.%",
```

@Lucifergene Can we use a more generic description here (and in the other dropdowns on that page) in a follow-up PR?

Something like:

```suggestion
        "description": "%devconsole~The default for the \"Deploy"\ resource type when a user import a component via \"Import from Git\", \"Import JAR file\" or \"Deploy Image\".%",
```